go/enclave/rpc_server.go

```diff
@@ -266,3 +271,27 @@ func (s *server) decodeBlock(encodedBlock []byte) types.Block {
 	}
 	return block
 }
+
+// serializeEVMError deserializes EVM errors into the RPC response
+// always returns a SerialisableError byte slice
+func serializeEVMError(err error) ([]byte, error) {
+	var errSerializedBytes []byte
+	var errReturn interface{}
+	var marshallErr error
+
+	// it's a generic error, can't extract more info
+	errReturn = evm.SerialisableError{Err: err.Error()}
+
+	// it's already a serialized error, handle any error wrapping that might have occurred
+	var e evm.SerialisableError
+	if ok := errors.As(err, &e); ok {
+		errReturn = e
+	}
+
+	// serialise the error object returned by the evm into a json
+	errSerializedBytes, marshallErr = json.Marshal(errReturn)
```

Comment:
doesn't this break the old behaviour where we were serialising the error directly, which was already a serialisableError?

Reply:
If I understand the question correctly, No, it's ok. We were creating a Serialsable error which implements the interface error and was past down here.
The 
```
if ok := errors.As(err, &e); ok {
		errReturn = e
	}
```
ensures that SerialsableError is still used as is.